pkg/utils/jwt.go

```diff
@@ -36,3 +34,29 @@ func GetStringFieldFromJWT(token string, field string) (string, error) {
 
 	return claimString, nil
 }
+
+// GetUsernameFromJWT returns the username extracted from JWT token
+func GetUsernameFromJWT(token string) string {
+	var jwtToken *jwt.Token
+	var err error
+	parser := new(jwt.Parser)
+	jwtToken, _, err = parser.ParseUnverified(token, jwt.MapClaims{})
+	if err != nil {
+		return "anonymous"
+	}
+	claims, ok := jwtToken.Claims.(jwt.MapClaims)
+	if !ok {
+		return "anonymous"
+	}
+	claim, ok := claims["username"]
+	if !ok {
+		return "anonymous"
+	}
+	return claim.(string)
+}
+
+// GetContextNickname returns a nickname of a context
+func GetContextNickname(namespace, clusterNick, userNick string) string {
+	tokens := strings.SplitN(userNick, "/", 2)
+	return namespace + "/" + clusterNick + "/" + tokens[0]
```

Comment:
not exactly sure, it will most likely break your local kubeconfig context for a cluster :D 
this is from existing login.go and i would avoid changing its behavior in this PR if at all possible
I added test cases anyway, but they mostly highlight the problem